spec/data-types-and-values.html

```diff
@@ -159,8 +159,25 @@ <h1>
 
     <emu-clause id="sec-ecmascript-language-types-box-type">
       <h1>The Box Type</h1>
-      <p>The Box type is the set of all the possible singleton primitive wrappers around any ECMAScript value. Each box value holds an associated [[Value]] containing an ECMAScript value. The [[Value]] internal slot is never modified.</p>
+      <p>The Box type is the set of all the possible singleton primitive wrappers around any ECMAScript value. Each box value holds an associated [[Value]] containing an ECMAScript value and a [[Realm]] containing the Realm of the Box function which created the Box value. The [[Value]] and [[Realm]] internal slots are never modified.</p>
 
+      <emu-clause id="sec-ecmascript-language-types-box-unbox" type="abstract operation">
+        <h1>
+          Unbox (
+            _box_: a Box,
+          )
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It extracts the boxed Object from _box_, throwing if it has been boxed by a different Realm's Box constructor.</dd>
+        </dl>
+        <emu-alg>
+          1. If Type(_box_.[[Value]]) is Object, then

```

Comment:
This reminds me, the `SameValue` check for `Box` need to only check the `[[Realm]]` slot if `Type([[Value]]) is Object`. I think the easiest to accomplish this would be to set `[[Realm]]` only if Object at construction of the `Box`, and always compare both slots.